featuretools/tests/computational_backend/test_pandas_backend.py

```diff
@@ -522,6 +572,17 @@ def test_empty_child_dataframe():
     assert_array_equal(fm2[names], [[0, np.nan]])
 
 
+def test_with_features_built_from_es_metadata(es):
```

Comment:
Yes, there was a bug where if you called `calculate_feature_matrix` with features which had been deserialized you would get different results. We had been doing something equivalent to `feature.relationship_path[0].entityset[entity_id].df`, but after deserialization the dataframe would always be empty. [The fix](https://github.com/Featuretools/featuretools/pull/572/commits/1a67817ad1d7a656a52a0247c3244c2cb051c741#diff-263657544b30befa38d6d1afcfe2e1c6R166) is to make sure we always get the dataframe from the entityset passed to `calculate_feature_matrix`.

EntitySet "metadata" which is  also an `EntitySet`, but without any data, can be confusing and seems likely to lead to more bugs like the above. I wonder if we could create a separate `EntitySetMetadata` class which wouldn't hold any dataframes. Features and relationships could reference this instead of `EntitySet` since they don't need the actual data.